packages/sysutils/busybox/scripts/init

```diff
@@ -81,9 +82,11 @@
     case $arg in
       BOOT_IMAGE=*)
         IMAGE_KERNEL="${arg#*=}"
+        IMAGE_KERNEL=$(basename $IMAGE_KERNEL)
```

Comment:
Yes, moved basename call to where is actually needed.
Regarding checking for block device: both calls could be removed because this is not actually used as I know - kernel.img must always be on first partition. I assume this is some old leftover which could be removed.